src/blockdata/transaction.rs

```diff
@@ -410,12 +409,24 @@ impl Transaction {
         script_pubkey: &Script,
         sighash_u32: u32
     ) -> SigHash {
+        if self.is_invalid_use_of_sighash_single(sighash_u32, input_index) {
+            return SigHash::from_slice(&UINT256_ONE).unwrap(); // Parse won't panic for const slice.
```

Comment:
Nit: we usually try to avoid any `unwraps` AFAIR:
```suggestion
            return SigHash::from_slice(&UINT256_ONE).expect("const-size array").
```

Reply:
Might as well fix it since we need to rebase anyway, but no, I don't particularly care either way.